_inc/lib/core-api/wpcom-endpoints/memberships.php

```diff
@@ -149,6 +198,50 @@ public function create_product( $request ) {
 		return $request;
 	}
 
+	/**
+	 * Automatically generate products according to type.
+	 *
+	 * @param object $request - request passed from WP.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function create_products( $request ) {
+		if ( ( defined( 'IS_WPCOM' ) && IS_WPCOM ) ) {
+			jetpack_require_lib( 'memberships' );
+			$connected_destination_account_id = Jetpack_Memberships::get_connected_account_id();
+			if ( ! $connected_destination_account_id ) {
+				return new WP_Error( 'no-destination-account', __( 'Please set up a Stripe account for this site first', 'jetpack' ) );
+			}
+			return Memberships_Product::generate_default_products( get_current_blog_id(), $request['type'], $request['currency'], $connected_destination_account_id );
+		} else {
+			$blog_id  = Jetpack_Options::get_option( 'id' );
+			$response = Client::wpcom_json_api_request_as_user(
+				"/sites/$blog_id/{$this->rest_base}/products",
+				'v2',
+				array(
+					'method' => 'POST',
+				),
+				array(
+					'type' => $request['type'],
+				)
+			);
+			if ( is_wp_error( $response ) ) {
+				if ( $response->get_error_code() === 'missing_token' ) {
+					return new WP_Error( 'missing_token', __( 'Please connect your user account to WordPress.com', 'jetpack' ), 404 );
+				}
+				return new WP_Error( 'wpcom_connection_error', __( 'Could not connect to WordPress.com', 'jetpack' ), 404 );
+			}
+			$data = isset( $response['body'] ) ? json_decode( $response['body'], true ) : null;
+			// If endpoint returned error, we have to detect it.
+			if ( 200 !== $response['response']['code'] && $data['code'] && $data['message'] ) {
+				return new WP_Error( $data['code'], $data['message'], 401 );
+			}
```

Comment:
Should we worry about returning a `WP_Error` even when some of those things (like `$data['message']`) are missing?

Reply:
Yeah, I think we can still return an error even if a message is set so I removed the `"data['message']` check in 2e7e2c4. I kept the `$data['code']` check though, since it's [required](https://core.trac.wordpress.org/browser/tags/5.3/src/wp-includes/class-wp-error.php#L55) by `WP_Error`.